modules/crypt.cpp

```diff
@@ -65,9 +199,17 @@ class CCryptMod : public CModule {
         AddCommand("ListKeys", static_cast<CModCommand::ModCmdFunc>(
                                    &CCryptMod::OnListKeysCommand),
                    "", "List all keys");
+        AddCommand("KeyX", static_cast<CModCommand::ModCmdFunc>(
+                                 &CCryptMod::OnKeyXCommand),
+                   "<Nick>", "Start a DH1080 key exchange with nick");
     }
 
-    ~CCryptMod() override {}
+    ~CCryptMod() override {
+        if (dh) {
+            DH_free(dh);
+            dh = NULL;
```

Comment:
But does std::unique_ptr know it should use DH_new() to create a new DH object and DH_free() to delete it?